src/appleseed.studio/mainwindow/rendering/lightpathswidget.cpp

```diff
@@ -160,149 +226,505 @@ void LightPathsWidget::slot_synchronize_camera()
+    Object* object = object_instance.find_object();
+
+    // This would already be logged in LightPathsWidget::load_scene_data
+    if (object == nullptr)
+        return;
+
+    const Transformd& transform = object_instance.get_transform();
+    const Matrix4f& object_transform_matrix(transform.get_local_to_parent());
+    const Matrix4f model_matrix = assembly_transform_matrix * object_transform_matrix;
+
+    // Object vertex buffer data has already been loaded; just add an instance
+    const string obj_name = string(object->get_name());
+    size_t buf_idx = m_scene_object_index_map.at(obj_name);
+
+    const GLuint object_instances_vbo = m_scene_object_instance_vbos[buf_idx];
+    const GLuint object_vao = m_scene_object_vaos[buf_idx];
+    const GLsizei current_instance = m_scene_object_current_instances[buf_idx];
+    m_scene_object_current_instances[buf_idx] += 1;
+
+    m_gl->glBindVertexArray(object_vao);
+    m_gl->glBindBuffer(GL_ARRAY_BUFFER, object_instances_vbo);
+    const Matrix4f gl_matrix = transpose(model_matrix);
+    m_gl->glBufferSubData(
+        GL_ARRAY_BUFFER,
+        current_instance * TransformByteStride,
+        TransformByteStride,
+        reinterpret_cast<const GLvoid*>(&gl_matrix[0]));
+}
 
-    glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+void LightPathsWidget::load_assembly_instance(
+    const AssemblyInstance& assembly_instance,
+    const float             time)
+{
+    const Assembly* assembly = assembly_instance.find_assembly();
 
-    static const GLfloat LightPosition[] = { 0.0f, 0.0f, 0.0f, 1.0f };
-    static const GLfloat LightAmbient[] = { 0.0f, 0.0f, 0.0f, 1.0f };
-    static const GLfloat LightDiffuse[] = { 0.8f, 0.8f, 0.8f, 1.0f };
-    static const GLfloat LightSpecular[] = { 0.3f, 0.3f, 0.3f, 1.0f };
+    // This would already be logged in LightPathsWidget::load_scene_data
+    if (assembly == nullptr)
+        return;
 
-    glEnable(GL_LIGHT0);
-    glLightfv(GL_LIGHT0, GL_POSITION, LightPosition);
-    glLightfv(GL_LIGHT0, GL_AMBIENT, LightAmbient);
-    glLightfv(GL_LIGHT0, GL_DIFFUSE, LightDiffuse);
-    glLightfv(GL_LIGHT0, GL_SPECULAR, LightSpecular);
-}
+    const Transformd transform = assembly_instance.transform_sequence().evaluate(time);
 
-void LightPathsWidget::resizeGL(int w, int h)
-{
-    glViewport(0, 0, static_cast<GLsizei>(w), static_cast<GLsizei>(h));
+    const Matrix4f transform_matrix(transform.get_local_to_parent());
+
+    for (const auto& object_instance : assembly->object_instances())
+        load_object_instance(object_instance, transform_matrix);
+
+    for (const auto& child_assembly_instance : assembly->assembly_instances())
+        load_assembly_instance(child_assembly_instance, time);
 }
 
-namespace
+void LightPathsWidget::load_object_data(const Object& object)
 {
-    void glMultMatrixd(const Matrix4d& m)
+    const string obj_name = string(object.get_name());
+    RENDERER_LOG_INFO("opengl: uploading mesh data for object: %s", obj_name.c_str());
```

Comment:
It seems like `RENDERER_LOG_DEBUG()` would be more appropriate here.

Same change for other similar info messages.